htdocs/emailcollector/class/emailcollector.class.php

```diff
@@ -3043,7 +3063,7 @@ public function doCollectOneCollector($mode = 0)
 							// Create ticket
 							$tickettocreate = new Ticket($this->db);
 
-							$alreadycreated = $tickettocreate->fetch(0, '', '', $msgid);
+							$alreadycreated = $tickettocreate->fetch(0, '', $trackid, $msgid);
```

Comment:
Do we really need to add $trackid ?
$msgid is already unique so if we found a ticket with the msgid, we know it is already recorded in database.

Reply:
Yes but the $msgid is never the same between the mails, so if a ticket is already created and we receive a mail that responds to the mail that had created the ticket then the new mail needs to not create a ticket and be linked to the ticket (Done with another operation in collector config)

Reply:
> Yes but the $msgid is never the same between the mails, so if a ticket is already created and we receive a mail that responds to the mail that had created the ticket then the new mail needs to not create a ticket and be linked to the ticket (Done with another operation in collector config)

I see.

Note that if the ticket was created with the msgid and trackid was not set, if we collect another email for the same trackid, we will not find the email with the fetch() because into the fetch, the test on trackid and msgid is a AND, and we need a OR.
I think we must do a
$alreadycreated1 = $tickettocreate->fetch(0, '', $trackid);
$alreadycreated2 = $tickettocreate->fetch(0, '', '', $msgid);
$alreadycreated = $alreadycreated1 + $alreadycreated2;



Bur we need OR.